webapp/src/js/bootstrapper/index.js

```diff
@@ -108,58 +109,50 @@
 
     const localMetaDb = window.PouchDB(getLocalMetaDbName(dbInfo, userCtx.name), POUCHDB_OPTIONS.local);
 
-    return Promise
-      .all([
-        initialReplicationLib.isReplicationNeeded(localDb, userCtx),
-        swRegistration,
-        setReplicationId(POUCHDB_OPTIONS, localDb)
-      ])
-      .then(([isInitialReplicationNeeded]) => {
-        utils.setOptions(POUCHDB_OPTIONS);
-
-        if (isInitialReplicationNeeded) {
-          const replicationStarted = performance.now();
-          // Polling the document count from the db.
-          return initialReplicationLib
-            .replicate(remoteDb, localDb)
-            .then(() => initialReplicationLib.isReplicationNeeded(localDb, userCtx))
-            .then(isReplicationStillNeeded => {
-              if (isReplicationStillNeeded) {
-                throw new Error('Initial replication failed');
-              }
-            })
-            .then(() => window.startupTimes.replication = performance.now() - replicationStarted);
-        }
-      })
-      .then(() => {
-        const purgeMetaStarted = performance.now();
-        return purger
-          .purgeMeta(localMetaDb)
-          .on('should-purge', shouldPurge => window.startupTimes.purgingMeta = shouldPurge)
-          .on('start', () => setUiStatus('PURGE_META'))
-          .on('done', () => window.startupTimes.purgeMeta = performance.now() - purgeMetaStarted)
-          .catch(err => {
-            console.error('Error attempting to purge meta db - continuing', err);
-            window.startupTimes.purgingMetaFailed = err.message;
-          });
-      })
-      .then(() => setUiStatus('STARTING_APP'))
-      .catch(err => err)
-      .then(err => {
-        localDb.close();
-        remoteDb.close();
-        localMetaDb.close();
-
-        if (err) {
-          const errorCode = err.status || err.code;
-          if (errorCode === 401) {
-            return redirectToLogin(dbInfo);
-          }
-          setUiError(err);
-          throw (err);
+    try {
+      const [isInitialReplicationNeeded] = await Promise
+        .all([
+          initialReplicationLib.isReplicationNeeded(localDb, userCtx),
+          swRegistration,
+          setReplicationId(POUCHDB_OPTIONS, localDb),
+          offlineDdocs.init(localDb)
```

Comment:
This is the only real change here. The rest is just a quick re-factor to use async/await.